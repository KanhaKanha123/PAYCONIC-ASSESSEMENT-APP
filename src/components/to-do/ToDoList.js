import { Fragment } from "react";

const ToDoList = ({ listToDo, deleteToDo, editToDo }) => {
    return (<Fragment>
        <section aria-label="to do list in table" className="to-do--list-container">
            {(listToDo || []).map((item, index) => <div data-testid="list-todo-component" className="to-do--list-inner-container" key={index}>
                <span aria-label="to do text" data-testid="list-todo-text-span" className="to-do--list-inner-item-one">{item.value}</span>
                <span aria-label="to do edit button" className="to-do--list-inner-item-two"><button data-testid="list-todo-edit-btn" onClick={() => editToDo(item)}>Edit</button></span>
                <span aria-label="to do delete button" className="to-do--list-inner-item-three"><button data-testid="list-todo-delete-btn" onClick={() => deleteToDo(item.id)}>Delete</button></span>
            </div>)}
        </section>
    </Fragment >)
}

export default ToDoList;