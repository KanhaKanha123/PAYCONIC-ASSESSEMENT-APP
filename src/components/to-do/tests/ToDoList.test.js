import { render, screen, fireEvent, act, waitFor } from '@testing-library/react';
import ToDoList from '../ToDoList';
import renderer from 'react-test-renderer';

describe('ToDoList component test', () => {
    const mockFnDeleteToDo = jest.fn();
    const mockFnEditToDo = jest.fn();
    const listToDo = [{ id: 0, value: 'check' }]
    test("ToDoList component load correctly", () => {

        render(<ToDoList listToDo={listToDo} deleteToDo={mockFnDeleteToDo} editToDo={mockFnEditToDo} />);

        const listToDoComponent = screen.getAllByTestId('list-todo-component');
        const listToDoTextSpan = screen.getAllByTestId('list-todo-text-span');
        const deleteBtn = screen.getAllByTestId('list-todo-delete-btn');
        const editBtn = screen.getAllByTestId('list-todo-edit-btn');

        // check element exist
        expect(listToDoComponent).toBeInTheDocument;
        expect(listToDoTextSpan).toBeInTheDocument;
        expect(deleteBtn).toBeInTheDocument;
        expect(editBtn).toBeInTheDocument;
    });

    test("deleteToDo method should be called", async () => {

        await act(() => { render(<ToDoList listToDo={listToDo} deleteToDo={mockFnDeleteToDo} editToDo={mockFnEditToDo} />) });

        const deleteBtn = screen.getAllByTestId('list-todo-delete-btn');

        waitFor(() => {
            fireEvent.click(deleteBtn, { id: 1, value: 'check' })
            expect(mockFnDeleteToDo).toHaveBeenCalled;
        });
    });

    test("editToDo method should be called", async () => {

        await act(() => { render(<ToDoList listToDo={listToDo} deleteToDo={mockFnDeleteToDo} editToDo={mockFnEditToDo} />) });

        const editBtn = screen.getAllByTestId('list-todo-edit-btn');

        waitFor(() => {
            fireEvent.click(editBtn, { id: 1, value: 'check' })
            expect(mockFnDeleteToDo).toHaveBeenCalled;
        });
    });

    test("ToDoList component should match snapshot", async () => {
        const element = await renderer.create(<ToDoList listToDo={listToDo} deleteToDo={mockFnDeleteToDo} editToDo={mockFnEditToDo} />);
        expect(element.toJSON()).toMatchSnapshot();
    });
});
