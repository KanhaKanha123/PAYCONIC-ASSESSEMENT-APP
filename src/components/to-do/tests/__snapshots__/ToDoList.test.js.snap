// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList component test ToDoList component should match snapshot 1`] = `
<section
  aria-label="to do list in table"
  className="to-do--list-container"
>
  <div
    className="to-do--list-inner-container"
    data-testid="list-todo-component"
  >
    <span
      aria-label="to do text"
      className="to-do--list-inner-item-one"
      data-testid="list-todo-text-span"
    >
      check
    </span>
    <span
      aria-label="to do edit button"
      className="to-do--list-inner-item-two"
    >
      <button
        data-testid="list-todo-edit-btn"
        onClick={[Function]}
      >
        Edit
      </button>
    </span>
    <span
      aria-label="to do delete button"
      className="to-do--list-inner-item-three"
    >
      <button
        data-testid="list-todo-delete-btn"
        onClick={[Function]}
      >
        Delete
      </button>
    </span>
  </div>
</section>
`;
